### PR TITLE
handle aws generated tags

### PR DESCRIPTION
**Pull request by Sean-Holcomb**
## What does this PR change?
This PR updates AWS CUR integration to ingest AWS generated tags in addition to user generated tags. To do this columns with the AWS label prefix are gathered in a separate slice. With this separate list of columns the prefix can be properly handled and saved to cloud cost labels. This PR includes this addition for both Athena and S3 integrations.

## Does this PR relate to any other PRs?
* 

## How will this PR impact users?
* Users will now be able to see AWS generated labels in Cloud Costs
Screenshot 2024-08-29 at 1 08 42 PM

## Does this PR address any GitHub or Zendesk issues?
* Closes ENG-150

## How was this PR tested?
* This PR was tested by running cloud cost and aggregator locally and ensuring that AWS label `aws_eks_cluster_name` was being populated in cloud cost.
`curl --location --globoff 'http://localhost:9004/cloudCost?window=yesterday&filter=label[aws_eks_cluster_name]%3A%22dev-1%22&aggregate=label%3Aaws_eks_cluster_name'`
```
      "cloudCosts": {
                    "dev-1": {
                        "properties": {
                            "labels": {
                                "aws_created_by": "AssumedRole:AROAUKXXWPGDYLPOEB2GE:AutoScaling",
                                "aws_eks_cluster_name": "dev-1",
                                "eks_cluster_name": "dev-1"
                            }
                        },
                        "window": {
                            "start": "2024-08-27T00:00:00Z",
                            "end": "2024-08-28T00:00:00Z"
                        },
                        "listCost": {
                            "cost": 28.5624505521,
                            "kubernetesPercent": 1
                        },
                        "netCost": {
                            "cost": 28.5624505521,
                            "kubernetesPercent": 1
                        },
                        "amortizedNetCost": {
                            "cost": 28.5624505521,
                            "kubernetesPercent": 1
                        },
                        "invoicedCost": {
                            "cost": 28.5624505521,
                            "kubernetesPercent": 1
                        },
                        "amortizedCost": {
                            "cost": 28.5624505521,
                            "kubernetesPercent": 1
                        }
                    }
                },
                "window": {
                    "start": "2024-08-27T00:00:00Z",
                    "end": "2024-08-28T00:00:00Z"
                },
                "aggregationProperties": [
                    "label:aws_eks_cluster_name"
                ]
            }
```

Because this PR deals with integrations it will need to by pass unit testing gates

## Does this PR require changes to documentation?
* 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

File: pkg/cloud/aws/athenaintegration.go [hunk 1]
```diff
@@ -347,6 +353,15 @@ func (ai *AthenaIntegration) RowToCloudCost(row types.Row, aqi AthenaQueryIndexe
 		}
 	}
 
+	for _, awsColumnName := range aqi.AWSTagColumns {
+		// partially remove prefix leaving "aws_"
+		labelName := strings.TrimPrefix(awsColumnName, "resource_tags_")
```
Comment (ameijer) [inline pkg/cloud/aws/athenaintegration.go]: define new const, or string.TrimSuffix your existing const

File: pkg/cloud/aws/s3selectintegration.go [hunk 2]
```diff
@@ -313,6 +331,9 @@ func (s3si *S3SelectIntegration) GetCloudCost(
 
 // hsK8sLabel checks if the labels contain a k8s label
 func hasK8sLabel(labels opencost.CloudCostLabels) bool {
+	if _, ok := labels["aws:eks:cluster-name"]; ok {
```
Comment (ameijer) [inline pkg/cloud/aws/s3selectintegration.go]: const?

File: pkg/cloud/aws/s3selectintegration.go [hunk 3]
```diff
@@ -178,6 +185,17 @@ func (s3si *S3SelectIntegration) GetCloudCost(
...
+			for _, awsLabelColumnName := range awsLabelColumns {
+				// remove quotes
+				labelName := strings.TrimPrefix(awsLabelColumnName, `s."`)
+				labelName = strings.TrimSuffix(labelName, `"`)
+				// partially remove prefix leaving "aws:"
+				labelName = strings.TrimPrefix(labelName, "resourceTags/")
+				value := GetCSVRowValue(row, columnIndexes, awsLabelColumnName)
+				if value != "" {
+					labels[labelName] = value
+				}
```
Comment (ameijer) [inline pkg/cloud/aws/s3selectintegration.go:198]: tests please!

File: pkg/cloud/aws/athenaintegration.go (hunk 1 above)
Comment (kaelanspatel) [reply pkg/cloud/aws/athenaintegration.go]: Could also just strip prefix in the first loop, and then append here.

**Review by kaelanspatel — APPROVED** [review]
Other than Alex's concerns, lgtm fwiw.

File: pkg/cloud/aws/s3selectintegration.go (hunk 3 above)
Comment (Sean-Holcomb) [reply pkg/cloud/aws/s3selectintegration.go:198]: These changes relate to methods that integrate with s3 and Athena. Unit tests are not applicable in this case.

**Review by ameijer — APPROVED** [review]
approving, but I think this is definitely unit testable

**Pull request merged**